shared/metrics/src/main/java/io/pravega/shared/metrics/StatsProviderImpl.java

```diff
@@ -66,84 +49,35 @@ public void start() {
-        }
         if (conf.isEnableStatsdReporter()) {
-            log.info("Configuring stats with statsD at {}:{}", conf.getStatsDHost(), conf.getStatsDPort());
-            reporters.add(StatsDReporter.forRegistry(getMetrics())
-                                        .prefixedWith(conf.getMetricsPrefix())
-                                        .build(conf.getStatsDHost(), conf.getStatsDPort()));
+            Metrics.addRegistry(new StatsdMeterRegistry(RegistryConfigUtil.createStatsdConfig(conf), Clock.SYSTEM));
         }
-        if (conf.isEnableGraphiteReporter()) {
-            log.info("Configuring stats with graphite at {}:{}", conf.getGraphiteHost(), conf.getGraphitePort());
-            final Graphite graphite = new Graphite(new InetSocketAddress(conf.getGraphiteHost(), conf.getGraphitePort()));
-            reporters.add(GraphiteReporter.forRegistry(getMetrics())
-                .prefixedWith(conf.getMetricsPrefix())
-                .convertRatesTo(TimeUnit.SECONDS)
-                .convertDurationsTo(TimeUnit.MILLISECONDS)
-                .filter(MetricFilter.ALL)
-                .build(graphite));
-        }
-        if (conf.isEnableJMXReporter()) {
-            log.info("Configuring stats with jmx {}", conf.getJmxDomain());
-            final JmxReporter jmx = JmxReporter.forRegistry(getMetrics())
-                .inDomain(conf.getJmxDomain())
-                .convertRatesTo(TimeUnit.SECONDS)
-                .convertDurationsTo(TimeUnit.MILLISECONDS)
-                .build();
-            jmx.start();
-        }
-        if (conf.isEnableGangliaReporter()) {
-            try {
-                log.info("Configuring stats with ganglia at {}:{}", conf.getGangliaHost(), conf.getGangliaPort());
-                final GMetric ganglia = new GMetric(conf.getGangliaHost(), conf.getGangliaPort(), GMetric.UDPAddressingMode.MULTICAST, 1);
-                reporters.add(GangliaReporter.forRegistry(getMetrics())
-                    .prefixedWith(conf.getMetricsPrefix())
-                    .convertRatesTo(TimeUnit.SECONDS)
-                    .convertDurationsTo(TimeUnit.MILLISECONDS)
-                    .build(ganglia));
-            } catch (IOException e) {
-                log.warn("ganglia create failure: {}", e);
-            }
+
+        if (conf.isEnableInfluxDBReporter()) {
+            Metrics.addRegistry(new InfluxMeterRegistry(RegistryConfigUtil.createInfluxConfig(conf), Clock.SYSTEM));
         }
-        if (conf.isEnableConsoleReporter()) {
-            log.info("Configuring console reporter");
-            reporters.add(ConsoleReporter.forRegistry(getMetrics())
-                .convertRatesTo(TimeUnit.SECONDS)
-                .convertDurationsTo(TimeUnit.MILLISECONDS)
-                .build());
+
+        if (Metrics.globalRegistry.getRegistries().size() == 0) {
+            log.error("Error! No concrete metrics register bound, the composite registry runs as no-op!");
```

Comment:
You're already logging at error level; no need to include `Error!` in your message.

Reply:
Fixed